src/pages/admin/register-xpub.jsx

```diff
@@ -1,66 +1,66 @@
 import bsv from 'bsv';
 import React, { useCallback, useEffect, useState } from 'react';
-import { useNavigate } from "react-router-dom";
+import { useNavigate } from 'react-router-dom';
 
-import { Alert, Button, TextField, Typography } from "@mui/material";
+import { Alert, Button, TextField, Typography } from '@mui/material';
 
-import { DashboardLayout } from "../../components/dashboard-layout";
-import { useUser } from "../../hooks/user";
-import logger from "../../logger";
+import { DashboardLayout } from '../../components/dashboard-layout';
+import { useUser } from '../../hooks/useUser';
+import logger from '../../logger';
 
 export const AdminRegisterXPub = () => {
   const navigate = useNavigate();
-  const { spvWalletAdminClient } = useUser();
+  const { spvWalletClient, admin } = useUser();
 
-  const [ newXPub, setNewXPub ] = useState("");
-  const [ xPrivInput, setXPrivInput ] = useState("");
-  const [ loading, setLoading ] = useState(false);
-  const [ error, setError ] = useState('');
+  const [newXPub, setNewXPub] = useState('');
+  const [xPrivInput, setXPrivInput] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
 
   useEffect(() => {
-    if (!spvWalletAdminClient) {
+    if (!admin) {
       navigate('/');
     }
-  }, [spvWalletAdminClient]);
+  }, [navigate, admin]);
 
   useEffect(() => {
     if (xPrivInput) {
       try {
         const xPrivHD = bsv.HDPrivateKey.fromString(xPrivInput); // will throw on error
         setNewXPub(xPrivHD.hdPublicKey.toString());
-        setXPrivInput("");
-      } catch(e) {
-          logger.error(e)
+        setXPrivInput('');
+      } catch (e) {
+        logger.error(e);
         setError(e.message);
       }
     }
   }, [xPrivInput]);
 
-  const handleRegisterXPub = useCallback((newXPub) => {
-    if (!newXPub) {
-      setError("No xPub to add")
-        logger.info("No xPub to add")
-    }
-    setLoading(true);
-    try {
-      const xPubHD = bsv.HDPublicKey.fromString(newXPub); // will throw on error
-        spvWalletAdminClient.AdminNewXpub(newXPub);
-      alert("XPub added");
-      logger.info("XPub added")
-      setNewXPub("");
-    } catch(e) {
-        logger.error(e)
-      setError(e.message);
-    }
-    setLoading(false);
-  }, [spvWalletAdminClient]);
+  const handleRegisterXPub = useCallback(
+    (newXPub) => {
+      if (!newXPub) {
+        setError('No xPub to add');
+        logger.info('No xPub to add');
+      }
+      setLoading(true);
+      try {
+        bsv.HDPublicKey.fromString(newXPub); // will throw on error
```

Comment:
Do we need this line? Maybe we should instead return an error from the line below `spvWalletClient.AdminNextXpub(newXPub);`? This way we can remove `bsv` lib

Reply:
As I mentioned previously:

> However, I have to admit I haven't inspected the whole code good enough...It seems there are other places in this project where the bsv package is used.

Given this, addressing this is likely out of scope for this specific PR.

Reply:
> This is probably a way to "validate" the string

Yes, I know, but maybe this validation could be done in `AdminNextXpub` method, because we want to get rid of bsv

> Given this, addressing this is likely out of scope for this specific PR

Got it